tests/test-sessions/test-infinispan-sessions/src/test/java/org/eclipse/jetty/server/session/remote/RemoteInfinispanTestSupport.java

```diff
@@ -31,42 +31,93 @@
 import org.infinispan.client.hotrod.marshall.ProtoStreamMarshaller;
 import org.infinispan.protostream.FileDescriptorSource;
 import org.infinispan.protostream.SerializationContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.Testcontainers;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.output.Slf4jLogConsumer;
+import org.testcontainers.containers.wait.strategy.LogMessageWaitStrategy;
+import org.testcontainers.utility.MountableFile;
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertTrue;
-import static org.junit.jupiter.api.Assertions.fail;
 
 /**
  * RemoteInfinispanTestSupport
  */
 public class RemoteInfinispanTestSupport
 {
+    private static final Logger LOG = LoggerFactory.getLogger(RemoteInfinispanTestSupport.class);
     public static final String DEFAULT_CACHE_NAME = "session_test_cache";
     public RemoteCache<String, SessionData> _cache;
     private String _name;
     public static RemoteCacheManager _manager;
+    private static final Logger INFINISPAN_LOG =
+        LoggerFactory.getLogger("org.eclipse.jetty.server.session.remote.infinispanLogs");
+
+    static GenericContainer infinispan;
 
     static
     {
         try
         {
-            String host = System.getProperty("hotrod.host", "127.0.0.1");
-
+            Testcontainers.exposeHostPorts(11222);
+            long start = System.currentTimeMillis();
+            String infinispanVersion = System.getProperty("infinispan.docker.image.version", "9.4.8.Final");
+            infinispan =
+                new GenericContainer(System.getProperty("infinispan.docker.image.name", "jboss/infinispan-server") +
+                                         ":" + infinispanVersion)
+                    .withEnv("APP_USER","theuser")
+                    .withEnv("APP_PASS","foobar")
+                    .withEnv("MGMT_USER", "admin")
+                    .withEnv("MGMT_PASS", "admin")
+                    .waitingFor(new LogMessageWaitStrategy()
+                                    .withRegEx(".*Infinispan Server.*started in.*\\s"))
+                    .withExposedPorts(4712,4713,8088,8089,8443,9990,9993,11211,11222,11223,11224)
+                    .withCopyFileToContainer(MountableFile.forClasspathResource("remote-session-test.xml"),
+                                              "/opt/jboss/infinispan-server/standalone/configuration/remote-session-test.xml")
+                    .withLogConsumer(new Slf4jLogConsumer(INFINISPAN_LOG));
+            // we could simply use `standalone` as well but... :)
+            infinispan =  infinispan.withCommand("-c remote-session-test.xml");
+            infinispan.start();
+            String host = infinispan.getContainerIpAddress();
+            System.setProperty("hotrod.host", host);
+            int port = infinispan.getMappedPort(11222);
```

Comment:
@janbartel 11222 is the default hotrod port and this is the one defined in the xml file. so yes we use hotrod :)